micro-rdk/src/common/grpc.rs

```diff
@@ -1499,11 +1564,68 @@ where
                 Some(path) => path.as_str(),
                 None => return Err(GrpcError::RpcInvalidArgument),
             };
-            svc.process_request(path, msg);
+
+            let grpc = GrpcServerInner {
+                buffer: Some(BytesMut::with_capacity(GRPC_BUFFER_SIZE)),
+                robot: &svc.robot,
+                signaling_server: &svc.signaling_server,
+            };
+
+            let mut trailers = HeaderMap::new();
+            trailers.insert("grpc-status", "0".parse().unwrap());
+            struct UnfoldState {
+                trailers: HeaderMap,
+                #[allow(clippy::type_complexity)]
+                stream: Option<
+                    Pin<
+                        Box<
+                            dyn futures_lite::Stream<Item = Result<Bytes, ServerError>>
+                                + Sync
+                                + Send,
+                        >,
+                    >,
+                >,
+            }
+            let mut state = UnfoldState {
+                trailers: trailers.clone(),
+                stream: None,
+            };
+
+            match grpc.validate_rpc(&msg).map_err(ServerError::from) {
+                Ok(payload) => {
+                    let _ = state.stream.insert(grpc.handle_request(path, payload));
+                }
+                Err(e) => {
+                    state.trailers.insert("grpc-status", e.status_code().into());
+                    state
+                        .trailers
+                        .insert("grpc-message", e.to_string().parse().unwrap());
+                }
+            };
+
+            let stream = futures_lite::stream::unfold(state, |mut state| async move {
```

Comment:
The `unfold` gadget is interesting: it lets you have a stateful stream. This might be better handled by implementing our own future on which we can `poll_frame` as GrpcBody does currently, but this was enough to get off the ground.